### ISSUE TITLE
output format

### ISSUE THREAD

**Issue by mimoo** [mimoo/cargo-specification#5]
I think cargo-specification should support:

* different backends to produce an output (zkdocs, respec, gitbook, mdbook, single-page RFC, noise-type spec, etc.)
* running commands to generated parts of the specification (for example we might want to run code in our codebase to generate LaTeX/markdown description of what is implemented)

this way it does not try to solve every aspect of the problem.

**Comment by ciresnave** [comment]
Agreed.  This also ties directly to [issue 13](https://github.com/mimoo/cargo-specification/issues/13).  A run-time pluggable list of output generators would be wonderful.  For that matter, the ability to use multiple output generators would be nice...Markdown, and a PDF for example.

Another thought related to this you may have already had:  

Should different templates be used for different outputs?  For example, I might want to output a developer doc in Markdown but also create an RFC for submission to the IETF.
